sklearn/manifold/t_sne.py

```diff
@@ -714,6 +714,11 @@ def _fit(self, X, skip_num_points=0):
             if self.verbose:
                 print("[t-SNE] Computing {} nearest neighbors...".format(k))
 
+            if(sp.issparse(X)):
+                if np.any(getnnz(X, axis=1) < k):
+                    raise ValueError("Perplexity of the sparse matrix is high."
```

Comment:
It is not the perplexity of the sparse matrix. The problem is that the requested tSNE perplexity is too high for the number of neighbours provided in the precomputed sparse matrix. It would be good if your error message also stated the number of neighbours required for the sought perplexity.